apps/mobile/src/features/accounts/account-selector/account-selector-sheet.layout.tsx

```diff
@@ -31,62 +31,61 @@ export function AccountSelectorSheetLayout({
-              direction={direction}
-              scrollViewRef={scrollViewRef}
-              placeholderIdx={placeholderIdx}
-              cardsLength={accounts.length}
-              key={account.id}
-              cardId={account.id}
-              onCardPress={() => onAccountPress(account.id)}
-              swapCardIndexes={swapAccountIndexes}
-            >
-              <WalletLoader fingerprint={account.fingerprint} key={account.id}>
-                {wallet => (
-                  <AccountCard
-                    address={
-                      <AccountAddress
-                        accountIndex={account.accountIndex}
-                        fingerprint={account.fingerprint}
-                      />
-                    }
-                    balance={
-                      <AccountBalance
-                        accountIndex={account.accountIndex}
-                        fingerprint={account.fingerprint}
-                      />
-                    }
-                    icon={
-                      <AvatarIcon
-                        color={theme.colors['ink.background-primary']}
-                        icon={account.icon}
-                      />
-                    }
-                    name={account.name}
-                    walletName={wallet.name}
-                    iconTestID={defaultIconTestId(account.icon)}
-                  />
-                )}
-              </WalletLoader>
-            </Draggable>
-          ))}
+          <Box gap="2">
+            {accounts.map((account, idx) => (
+              <Draggable
+                idx={idx}
+                direction={direction}
+                scrollViewRef={scrollViewRef}
+                placeholderIdx={placeholderIdx}
+                cardsLength={accounts.length}
+                key={account.id}
+                cardId={account.id}
+                onCardPress={() => onAccountPress(account.id)}
+                swapCardIndexes={swapAccountIndexes}
+              >
+                <WalletLoader fingerprint={account.fingerprint} key={account.id}>
+                  {wallet => (
+                    <AccountCard
+                      caption={wallet.name}
+                      primaryTitle={account.name}
+                      secondaryTitle={
+                        <AccountBalance
+                          variant="label01"
+                          accountIndex={account.accountIndex}
+                          fingerprint={account.fingerprint}
+                        />
+                      }
+                      address={
+                        <AccountAddress
+                          accountIndex={account.accountIndex}
+                          fingerprint={account.fingerprint}
+                        />
+                      }
+                      icon={account.icon}
+                      iconTestID={defaultIconTestId(account.icon)}
+                    />
+                  )}
+                </WalletLoader>
```

Comment:
I see this was here before, but isn't depending on a WalletLoader (and thus state), beyond the responsibility of a layout component?

Shouldn't everything within the loader be passed from the parent?